### DISCUSSION TITLE
Action iconButton: color() works if ->url() is specified. If ->action() the icon is always white.

### DISCUSSION THREAD

**Discussion by bvicini** [filamentphp/filament#14301] [Help]
### Package

Table builder

### Package Version

v3.2.122

### How can we help you?

Hello. For some reason I cannot find, when I set a table record action as iconButton, the color I specify with ->color() is met only if ->url() is invoked, while invoking ->action() I always get a white icon. If I remove ->iconButton() call, all works fine. Does someone have a hint of the reason why? Ty in advance

**Comment by wdog** [comment]
do not use ` ->iconButton()`

instead

```php
->hiddenLabel()
```

**Reply by bvicini** [reply]
ok thank you I will try that
